File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        vector<int> res;
        for (int i = 0; i < nums.size(); i++){
            int missing = target - nums[i];
            if (mp.find(missing) != mp.end()){
                
                res.push_back(mp[missing]);
                res.push_back(i);
            }
            mp[nums[i]] = i;
        }
        
        return res;
    }
};

File: 1-two-sum/1-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        dct = {}
        for i in range(len(nums)):
            if target - nums[i] in dct:
                return [i, dct[target - nums[i]]]
            dct[nums[i]] = i
        